### ISSUE TITLE
✨ Support `--depth` when using `dag.git`

### ISSUE THREAD

**Issue by Xiot** [dagger/dagger#7637]
### What are you trying to do?

Trying to pull down more history for a ref than just the single commit.

The request is to have a property on the `dag.git` to specify the `depth` (number of commits) to checkout

### Why is this important to you?

We currently have a couple different jobs within our workflow that rely on checking the differences between the given commit and its parent.

### How are you currently working around this?

Right now, we are using `dag.git` to do the initial checkout, then running a separate `git fetch --depth=2`, to pull the history.

For some reason, the separate fetch doesn't always pull down the history properly.

**Comment by jedevc** [comment]
Yup, this seems very reasonable to me, in my head this is strongly related to all the API improvements in https://github.com/dagger/dagger/issues/8520#issue-2539515747.

As a note to anyone picking this up - the logic for this needs to end up in our custom buildkit git source here: https://github.com/dagger/dagger/blob/d4d759140e4964437f9468d6537ff7a9fd313837/engine/sources/gitdns/source.go#L516-L523

There are *two* git repositories involved in a clone - the "farm" which keeps track of most of the repository, and then the clone off of the farm to get the git dir that the user sees. When specifying a depth, we need to ensure that the "farm" has at least the required depth, and then we just need to set the depth correctly when cloning off of the farm. The rest of this, plumbing through from the dagger API should be pretty easy.

**Comment by jedevc** [comment]
Fixed by https://github.com/dagger/dagger/pull/9980.
